geode-core/src/main/java/org/apache/geode/management/internal/operations/OperationExecutor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.operations;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import org.apache.geode.management.internal.api.LocatorClusterManagementService;
+import org.apache.geode.management.operation.OperationResult;
+
+/**
+ * Interface which any Management service operation must implement. {@code OperationExecutor}s are
+ * registered in the {@link LocatorClusterManagementService}
+ */
+public interface OperationExecutor {
+
```

Comment:
the name makes it sound like the thing that manages the queue of jobs, but it looks like it actually represents an individual job... maybe `ExecutableOperation` would be clearer?